infra/datalake/api_stack.py

```diff
@@ -0,0 +1,42 @@
+"""
+Data Lake AWS resources definitions.
+"""
+from aws_cdk import aws_lambda, core
+from aws_cdk.core import Tags
+
+
+class APIStack(core.Stack):
+    """Data Lake stack definition."""
+
+    def __init__(self, scope: core.Construct, stack_id: str, datasets_table, **kwargs) -> None:
+        super().__init__(scope, stack_id, **kwargs)
+
+        ############################################################################################
+        # ### API ENDPOINTS ########################################################################
+        ############################################################################################
+
+        endpoints = ("datasets",)
```

Comment:
YAGNI - we can reintroduce the looping if/when we need it, to keep the code as simple as possible.

Reply:
This change is not related to stack splitting. I can do this in other PR.